python/grass/jupyter/utils.py

```diff
@@ -87,6 +87,156 @@ def reproject_region(region, from_proj, to_proj):
+    :return: reprojected coordinates (returned as tuple)
+    """
+    # Prepare the input coordinate string
+    coord_str = f"{coord[1]} {coord[0]}\n"
+
+    # Start the m.proj command
+    proc = gs.start_command(
+        "m.proj",
+        input="-",
+        flags="i",
+        separator=",",
+        stdin=gs.PIPE,
+        stdout=gs.PIPE,
+        stderr=gs.PIPE,
+    )
+
+    proc.stdin.write(gs.encode(coord_str))
+    proc.stdin.close()
+    proc.stdin = None
+    proj_output, _ = proc.communicate()
+
+    output = gs.decode(proj_output).splitlines()
+    east, north, elev = map(float, output[0].split(","))
+
+    return east, north, elev
+
+
+def query_raster(coord, raster_list):
+    """Queries Raster
+
+    :param coord: coordinates given as tuple
+    :param list raster_list: list of raster names
+
+    :return str: formatted output of raster query results
+    """
+    output_list = [
+        """<table border='1'
+        style='border-collapse:collapse;
+        width: 100%;
+        font-size: 12px;'>"""
+    ]
+
+    for raster in raster_list:
+        raster_output = gs.raster.raster_what(map=raster, coord=coord)
+
+        output = f"""<tr style='background-color: #7CFC00;'>
+        <th colspan='3'>Raster: {raster}</th></tr>"""
+        if raster in raster_output[0] and "value" in raster_output[0][raster]:
+            value = raster_output[0][raster]["value"]
+            output += f"""
+            <tr style='background-color: #f0f0f0;'>
+            <td>1</td>
+            <td>Value</td>
+            <td>{value}</td>
+            </tr>
+            """
+
+        output_list.append(output)
+
+    if len(output_list) == 1:
+        return ""
+
+    output_list.append("</table>")
+    output_list.append("<br>")
+    final_output = "".join(output_list)
+    return final_output
+
+
+def query_vector(coord, vector_list, distance):
+    """Queries Vector
+
+    :param coord: coordinates given as tuple
+    :param list vector_list: list of vector names
+
+    :return str: formatted output of vector query results
+    """
+    output_list = [
+        """<table border='1'
+        style='border-collapse:collapse;
+        width: 100%;
+        font-size: 12px;'>"""
+    ]
+
+    for vector in vector_list:
+        vector_output = gs.vector.vector_what(
+            map=vector, coord=coord, distance=distance
+        )
+
+        if len(vector_output[0]) > 2:
+            output_list.append(
+                f"""
+                               <tr style='background-color: #7CFC00;'>
+                               <th colspan='3'>
+                               Vector: {vector}
+                               </th>
+                               </tr>
+                               """
+            )
+
+            items = list(vector_output[0].items())
+
+            for i, (key, value) in enumerate(items[1:], 1):
+                output = ""
+
+                if isinstance(value, dict):
```

Comment:
Let's not use the nested table with attributes. Let's do something like this:
![Selection_295](https://github.com/OSGeo/grass/assets/7494312/6844ea3f-e3c1-4bd2-95e0-baa64a254ef5)

So if there are no attributes, it will show just the Category and Layer. If there are attributes, it will show the attributes first.

Reply:
Please keep the Category and Layer, not the rest (like Driver, Database etc.)

Reply:
Okay, I have done one more thing here: the attribute will only be displayed if its value is None. Does this make sense?